### PR TITLE
Add feature to fixup code when user transposes 'record' in a declaration

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/59622

Looks like:

image

### CODE REVIEW

File: src/Analyzers/CSharp/CodeFixes/TransposeRecordKeyword/CSharpTransposeRecordKeywordCodeFixProvider.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                        recordDeclaration,
+                        (current, _) =>
+                        {
+                            var currentRecordDeclaration = (RecordDeclarationSyntax)current;
+                            if (!TryGetTokens(currentRecordDeclaration, out var recordKeyword, out var classOrStructKeyword, out var whitespace))
+                                return currentRecordDeclaration;
+
+                            return currentRecordDeclaration
+                                .WithClassOrStructKeyword(classOrStructKeyword.WithAppendedTrailingTrivia(whitespace))
+                                .WithKeyword(recordKeyword.WithLeadingTrivia(
```
Comment (davidwengier) [inline src/Analyzers/CSharp/CodeFixes/TransposeRecordKeyword/CSharpTransposeRecordKeywordCodeFixProvider.cs:117]: Should trailing trivia on `recordKeyword` be transferred to `classOrStructKeyword`?

File: src/Analyzers/CSharp/CodeFixes/CSharpCodeFixesResources.resx [hunk 2]
```diff
@@ -144,4 +144,7 @@
   <data name="Place_colon_on_following_line" xml:space="preserve">
     <value>Place colon on following line</value>
   </data>
+  <data name="Fix_token_placement" xml:space="preserve">
+    <value>Fix token placement</value>
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/CodeFixes/CSharpCodeFixesResources.resx:148]: sounds good to me :)

File: src/Analyzers/CSharp/CodeFixes/TransposeRecordKeyword/CSharpTransposeRecordKeywordCodeFixProvider.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/CodeFixes/TransposeRecordKeyword/CSharpTransposeRecordKeywordCodeFixProvider.cs:117]: yes.  that would atually make things substantially easier.  effectively we shuld keep all teh trivia fixed, and just swap the tokens (without trivia).  thanks!

**Comment by davidwengier** [comment]
Very nice :)

**Comment by jcouv** [comment]
Rebased since https://github.com/dotnet/roslyn/pull/59622 was just merged/squashed. It went smoothly (did `git rebase -i dotnet/main` and just deleted the first four commits from that previous PR; merges didn't cause any issues).
I verified things still build.

**Pull request merged**